target/linux/ramips/dts/mt7621_iptime_a8004t.dts

```diff
@@ -0,0 +1,137 @@
+/dts-v1/;
+
+#include "mt7621.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "iptime,a8004t", "mediatek,mt7621-soc";
+	model = "ipTIME A8004T";
+
+	chosen {
+		bootargs = "console=ttyS0,57600";
+	};
+
+	aliases {
+		led-boot = &led_cpu;
+		led-failsafe = &led_cpu;
+		led-running = &led_cpu;
+		led-upgrade = &led_cpu;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_cpu: cpu {
+			label = "a8004t:orange:cpu";
+			gpios = <&gpio0 18 GPIO_ACTIVE_LOW>;
+		};
+
+		led_wlan2g: wlan2g {
+			label = "a8004t:orange:wlan2g";
+			gpios = <&gpio0 17 GPIO_ACTIVE_LOW>;
+		};
+
+		led_wlan5g: wlan5g {
+			label = "a8004t:orange:wlan5g";
+			gpios = <&gpio0 14 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	button {
+		compatible = "gpio-keys-polled";
+		poll-interval = <20>;
+
+		reset {
+			label = "reset";
+			gpios = <&gpio0 4 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&gpio0 3 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+};
+
+&xhci {
+	status = "okay";
+};
+
+&spi0 {
+	status = "okay";
+
+	m25p80@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x20000>;
+				read-only;
+			};
+
+			partition@20000 {
+				label = "u-boot-env";
+				reg = <0x20000 0x10000>;
+				read-only;
+			};
+
+			factory: partition@30000 {
+				label = "factory";
+				reg = <0x30000 0x10000>;
+				read-only;
+			};
+
+			partition@40000 {
+				compatible = "denx,uimage";
+				label = "firmware";
+				reg = <0x40000 0xfc0000>;
+			};
+		};
+	};
+};
+
+&pcie {
+	status = "okay";
+};
+
+&pcie0 {
+	wifi@0,0 {
+		compatible = "mediatek,mt76";
+		reg = <0x0000 0 0 0 0>;
+		mediatek,mtd-eeprom = <&factory 0x0000>;
+		ieee80211-freq-limit = <2400000 2500000>;
+	};
+};
+
+&pcie1 {
+	wifi@0,0 {
+		compatible = "mediatek,mt76";
+		reg = <0x0000 0 0 0 0>;
+		mediatek,mtd-eeprom = <&factory 0x8000>;
+		ieee80211-freq-limit = <5000000 6000000>;
+	};
+};
+
+&ethernet {
+//	mtd-mac-address = <&factory 0xe000>;
```

Comment:
Found the correct one already? What about WAN?

What is the MAC address assignment of the stock firmware?

Reply:
I'm failed to find the correct one from EEPROM. I guess it is derived by its stock firmware...

Reply:
Well, it would be best to dump both factory and u-boot partitions with hexdump:
cat /proc/mtd
(Take the correct mtdX from the list)
hexdump -C /dev/mtdX

Then take one of the bytes from the MAC addresses vendor part and search the code for it.

You will then obtain a list of available addresses.

Then, you should have obtained an overview of addresses used on stock firmware (both Wifi, LAN, WAN). Look which of the addresses are present in flash and then calculate the ones missing ...

Please provide this overview here then. If you do not want to share your addresses, it's enough to give the last byte, e.g.
lan: *:0A (factory 0xe000, label)
wan: *:0B (factory 0xe006)
wifi2: *:0A (art 0x4)
wifi5: *:0C (art 0x4 + 2)

(This info could then be added to your commit description)

Reply:
My router has the following mac addresses:

WLAN(5G): <EFMNetworks>:60:6D:EE
WLAN(2.4G): <EFMNetworks>:60:6D:EC
WAN: <EFMNetworks>:60:6D:ED
LAN: <EFMNetworks>:60:6D:EC

Strange enough, LAN and WLAN1 shares MAC address. IDK where WAN address came from.

Reply:
Anyway, fixed.

Reply:
It seems the MAC setup of this device is same as that of A3. Are you sure the WAN MAC address is not in the flash? It's recommended to use address loaded from flash rather than calculated one. https://github.com/openwrt/openwrt/blob/master/target/linux/ramips/base-files/etc/board.d/02_network#L676

Reply:
Shared MAC of one WLAN and LAN is actually frequent. Nevertheless, frequently still two different flash locations are used.
So, I assume WiFi MAC addresses are correct?
2.4GHz *:EC 0x4
5 GHz *:EE 0x8004

So, if a hexdump of u-boot does not reveal any new addresses, we might have to stick with ethernet set from 0x4, too.

Reply:
I'll test this when I back home.

Reply:
I found both of them in u-boot image @ 0x1FC20 and 0x1FC40. 

Fixed.